post-processor/vagrant/azure.go

```diff
@@ -0,0 +1,67 @@
+package vagrant
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/packer/packer"
+)
+
+type AzureProvider struct{}
+
+func (p *AzureProvider) KeepInputArtifact() bool {
+	return true
+}
+
+func (p *AzureProvider) Process(ui packer.Ui, artifact packer.Artifact, dir string) (vagrantfile string, metadata map[string]interface{}, err error) {
+	// Create the metadata
+	metadata = map[string]interface{}{"provider": "azure"}
+
+	var AzureImageProps map[string]string
+	AzureImageProps = make(map[string]string)
+
+	// HACK(double16): It appears we can not access the Azure Artifact directly, so parse String()
```

Comment:
Is this because of the Azure builder? What would we/I need to make this right?

Reply:
I'm new to golang, so mostly I've been inspecting how other code is working. I tried
 a type assertion (? cast) but it didn't work, but I think I did it wrong. I found this example:

https://github.com/hashicorp/packer/blob/f79df4c440c7484e35db9efec0ea796d95e58afd/post-processor/vagrant/hyperv.go#L28-L31

If you could give me code to do the proper type assertion I can take it from there.

Reply:
Sorry, I should have looked at the other examples before I commented, but I did not.  After re-reading the other examples, it seems like this is the pattern being used.